src/tests/ftest/util/AgentUtils.py

```diff
@@ -127,31 +127,32 @@ def run_agent(basepath, server_list, client_list=None):
         sessions[client] = p
 
     # double check agent launched successfully
-    file_desc = sessions[client].stderr.fileno()
-    flags = fcntl.fcntl(file_desc, fcntl.F_GETFL)
-    fcntl.fcntl(file_desc, fcntl.F_SETFL, flags | os.O_NONBLOCK)
-    timeout = 5
-    start_time = time.time()
-    result = 0
-    pattern = "Starting daos_agent"
-    expected_data = ""
-    while True:
-        output = ""
-        try:
-            output = sessions[client].stderr.read()
-        except IOError as excpn:
-            if excpn.errno != errno.EAGAIN:
-                raise excpn
-            continue
-        match = re.findall(pattern, output)
-        expected_data += output
-        result += len(match)
-        if not output or time.time() - start_time > timeout:
-            print("<AGENT>: {}".format(expected_data))
-            raise AgentFailed("DAOS Agent didn't start!")
-        break
-    print("<AGENT> agent started and took %s seconds to start" % \
-            (time.time() - start_time))
+    for client in client_list:
+        file_desc = sessions[client].stderr.fileno()
+        flags = fcntl.fcntl(file_desc, fcntl.F_GETFL)
+        fcntl.fcntl(file_desc, fcntl.F_SETFL, flags | os.O_NONBLOCK)
+        timeout = 5
+        start_time = time.time()
+        result = 0
+        pattern = "Starting daos_agent"
+        expected_data = ""
+        while True:
+            output = ""
+            try:
+                output = sessions[client].stderr.read()
+            except IOError as excpn:
+                if excpn.errno != errno.EAGAIN:
+                    raise excpn
+                continue
```

Comment:
Should it sleep for a second or two?  This would be kind of a tight loop wouldn't it, constantly retrying and eating CPU.  

If it keeps returning no data then it will continue and the timeout logic will never get triggered and the loop never exits or at least runs longer than the timeout.  The timeout should be in the while condition so its guaranteed to get checked.

Reply:
Added in a 1sec sleep per loop. I also moved the timeout check above the continue, but left it as a discrete check vs. putting in the While just so we can have a unique message for the case of timeout vs. any other error.